metals/src/main/scala/scala/meta/internal/metals/BuildTargets.scala

```diff
@@ -649,4 +504,256 @@ object BuildTargets {
       leaves: collection.Set[BuildTargetIdentifier]
   )
 
+  final case class DataSeq(list: List[TargetData]) {
+    def iterator: Iterator[TargetData] = list.iterator
+    def writableDataIterator: Iterator[TargetData] = list.iterator
+    def iterable: Iterable[TargetData] = list.toIterable
+
+    def fromIterators[T](f: TargetData => Iterator[T]): Iterator[T] =
+      iterator.flatMap(f)
+    def fromOptions[T](f: TargetData => Option[T]): Option[T] =
+      fromIterators(f(_).iterator).take(1).toList.headOption
+  }
+
+  def createData(): TargetData =
+    new DataImpl
+
+  private final class DataImpl extends TargetData {
```

Comment:
Do we still need a separate `DataImpl` if it's just really a single class using this interface? I would just move everything to TargetData and make relevant fields private

Reply:
Should be fixed.